internal/provider/resource_gitlab_project_runner_enablement_test.go

```diff
@@ -0,0 +1,90 @@
+package provider
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+func TestAccGitlabProjectRunnerEnablement_basic(t *testing.T) {
+	testAccCheck(t)
+	testGroup := testAccCreateGroups(t, 1)[0]
+	projectA := testAccCreateProjectWithNamespace(t, testGroup.ID)
+	projectB := testAccCreateProjectWithNamespace(t, testGroup.ID)
+
+	name := fmt.Sprintf("TestAcc Runner %s", acctest.RandString(10))
+
+	opts := gitlab.RegisterNewRunnerOptions{
+		Token:       &projectA.RunnersToken,
+		Description: &name,
+	}
+
+	// Create runner in project A
+	runner, _, _ := testGitlabClient.Runners.RegisterNewRunner(&opts)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckGitlabProjectRunnerEnablementDestroy(projectB.ID, runner.ID),
+		Steps: []resource.TestStep{
+			// Attach it to projectB
+			{
+				Config: testAccGitlabProjectRunnerEnablementConfig(projectB.ID, runner.ID),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("gitlab_project_runner_enablement.foo", "project", fmt.Sprint(projectB.ID)),
+					resource.TestCheckResourceAttr("gitlab_project_runner_enablement.foo", "runner_id", fmt.Sprint(runner.ID)),
+					resource.TestCheckResourceAttr("gitlab_project_runner_enablement.foo", "id", fmt.Sprintf("%d:%d", projectB.ID, runner.ID)),
+					testAccCheckGitlabProjectRunnerEnablementCreate(projectB.ID, runner.ID),
+				),
+			},
```

Comment:
Let's add another test step testing what happens when the runner is attached to another project.

Reply:
I've added another step which adds an extra resource, and then verifies both resources

Reply:
The new test step is okay, however I was thinking about testing the recreation of a resource when the project changes - basically reassigning to another project. WDYT?

Reply:
Sure, I can add that

Reply:
Done